internal/supervisor/import.go

```diff
@@ -99,9 +105,96 @@ func NewImportSupervisor(awsClient toolsAWS.AWS, awat AWATClient, store installa
 	}
 }
 
+// completeImports transitions Installations back to stable after
+// checking with the AWAT to make sure that it has detected the
+// completion of the import
+func (s *ImportSupervisor) completeImports() error {
+	installationList, err := s.store.GetInstallations(
+		&model.InstallationFilter{
+			Paging: model.AllPagesNotDeleted(),
+			State:  model.InstallationStateImportComplete,
+		}, false, false)
+	if err != nil {
+		return errors.Wrap(err, "failed to list Installations")
+	}
+
+	for _, installation := range installationList {
+		importStatusList, err := s.awatClient.GetImportStatusesByInstallation(installation.ID)
+		if err != nil {
+			s.logger.WithError(err).Warnf("failed to get Import status for Installation %s", installation.ID)
+			continue
+		}
+		if len(importStatusList) == 0 {
+			s.logger.Warnf("no imports found for Installation with state %s, %s", installation.State, installation.ID)
+			continue
+		}
+		// find the most recently completed Import; that's the only one we
+		// care about
+		var mostRecentImport *awat.ImportStatus
+		for _, is := range importStatusList {
+			if is.State != awat.ImportStateComplete {
+				// An import is still running against this Installation.
+				// Multiple parallel imports against an Installation are not supported.
+				break
+			}
+			if mostRecentImport == nil || is.CompleteAt > mostRecentImport.CompleteAt {
+				mostRecentImport = is
+			}
+		}
+		if mostRecentImport == nil {
+			// an Import might still be running
+			continue
+		}
+
+		locked, err := s.store.LockInstallation(installation.ID, s.ID)
```

Comment:
This probably is not very likely, but I think we should refetch the Installation after lock (or possibly lock them all at the beginning of `completeImports`?) as some other Provisioner might have performed some change to it and we might revert it, or if we running 2 instances of Provisioner for imports weird race condition may happen here.

Reply:
Yeah, I am not sure that we are perfect with it, but refetching locked resources is definitely the approved method of working on resources.